src/lib/datepicker/calendar.ts

```diff
@@ -64,6 +77,13 @@ import {Directionality} from '@angular/cdk/bidi';
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class MatCalendar<D> implements AfterContentInit, OnDestroy, OnChanges {
+
+  /** An input indicating the type of the custom header component, if set. */
+  @Input() customCalendarHeaderComponent: ComponentType<any>;
+
+  /** A portal containing the header for this calendar. */
+  calendarHeaderPortal: Portal<any>;
```

Comment:
name this with a leading `_` to indicate internal

Reply:
done in 3467c0ac390cb4401cf7c44b79c1280ba5dcb5f8

I renamed it as you suggested, made it private and added a getter method since the template needs to access it. To my understanding, private members are not accessible from the component's template (although this would make sense, conceptually speaking): https://stackoverflow.com/questions/34574167/angular2-should-private-variables-be-accessible-in-the-template?rq=1

Reply:
Rather than a getter method just name it with an `_` and leave it public. We use this convention throughout Angular Material to mean "we had to leave this public due to technical constraints, but you shouldn't mess with it". We also automatically strip `_` properties when generating our docs.

Reply:
Got it!

done in b6938f24976978941d72e98b3567a847e49f42be